x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/index.tsx

```diff
@@ -34,19 +35,62 @@ const IndicesDetailsComponent: React.FC<Props> = ({
   const { patternRollups, patternIndexNames } = useResultsRollupContext();
   const { patterns } = useDataQualityContext();
 
+  const [isTourActive, setIsTourActive] = useState<boolean>(() => {
+    const isActive = localStorage.getItem(HISTORICAL_RESULTS_TOUR_IS_ACTIVE_STORAGE_KEY);
+    return isActive !== 'false';
+  });
+
+  const handleDismissTour = useCallback(() => {
+    setIsTourActive(false);
+    localStorage.setItem(HISTORICAL_RESULTS_TOUR_IS_ACTIVE_STORAGE_KEY, 'false');
+  }, []);
+
+  const [openPatterns, setOpenPatterns] = useState<Array<{ name: string; isOpen: boolean }>>(() => {
+    return patterns.map((pattern) => ({ name: pattern, isOpen: true }));
+  });
+
+  const handleAccordionToggle = useCallback((patternName: string, isOpen: boolean) => {
+    setOpenPatterns((prevOpenPatterns) => {
+      return prevOpenPatterns.map((p) => (p.name === patternName ? { ...p, isOpen } : p));
+    });
+  }, []);
+
+  const firstOpenPattern = useMemo(
+    () => openPatterns.find((pattern) => pattern.isOpen)?.name,
+    [openPatterns]
+  );
+
   return (
     <div data-test-subj="indicesDetails">
-      {patterns.map((pattern) => (
-        <StyledPatternWrapperFlexItem grow={false} key={pattern}>
-          <Pattern
-            indexNames={patternIndexNames[pattern]}
-            pattern={pattern}
-            patternRollup={patternRollups[pattern]}
-            chartSelectedIndex={chartSelectedIndex}
-            setChartSelectedIndex={setChartSelectedIndex}
-          />
-        </StyledPatternWrapperFlexItem>
-      ))}
+      {useMemo(
```

Comment:
NIT: I don't think this memoization is strictly necessary (I am not saying it's wrong either). But in these cases, I tend to prefer using `React.memo()` in a separate small pure component, than having a `useMemo` that returns React elements. Both options work though, just my five cents.

Reply:
Yes in most files it's not strictly necessary, but if you add all those strictly unnecessary instances across entire codebase it might become strictly necessary :D But I don't have a proof unless measured, so I agree with you. Especially if now I know there is no specific convention for that. I will remove useMemo's from this file, it will improve on readability a bit.

Reply:
and with regards to splitting away, I see your point, but this approach can lead to oversplitting and deepening of the render tree. One of the problems that I dealt with during refactoring ecs was exactly that, the unnecessarily splitting of components for the sake of purity :| I flattened them and it's become arguably easier to work with but it cost a gigantic effort of moving folders and imports around and merging tests etc... So it can be a double edged is what I mean, there is always a big unobvious cost.

Reply:
Done